doc/user_guide/recordflux_ug/03-getting_started_with_recordflux.rst

```diff
@@ -0,0 +1,726 @@
+         and Well_Formed (Ctx, F_Value)
+         and not Valid (Ctx, F_Value)
+      then
+         Ada.Text_IO.Put_Line ("Expected");
+      else
+         Ada.Text_IO.Put_Line ("Unexpected");
+      end if;
+
+      --  Inspect message contents
+      Take_Buffer (Ctx, Buffer);
+      if Buffer.all = (1, 0, 4, 10, 20, 30, 40) then
+         Ada.Text_IO.Put_Line ("Expected");
+      else
+         Ada.Text_IO.Put_Line ("Unexpected");
+      end if;
+   end Generate_TLV;
+
+Note that, as an opaque field, the ``Value`` field is
+well-formed but not valid (lines 30 and 31),
+since it has not been refined.
+
+Executing the program produces the following output:
+   
+.. code-block:: sh
+
+   $ ./generate_tlv
+   Expected
+   Expected
+   Expected
+
+Parsing a Message
+----------------------
+
+The following program decomposes a TLV message into its
+constituent fields:
+
+.. code:: ada
+   :number-lines:
+
+   with Ada.Text_IO; use Ada.Text_IO;
+   with RFLX.RFLX_Types;
+   with RFLX.RFLX_Builtin_Types;
+   with RFLX.TLV.Message; use RFLX.TLV.Message;
+   procedure Parse_TLV is
+
+      Buffer  : RFLX.RFLX_Builtin_Types.Bytes_Ptr := 
+        new RFLX.RFLX_Builtin_Types.Bytes'(1, 0, 4, 10, 20, 30, 40);
+      Ctx       : Context;
+      Tag       : RFLX.TLV.Tag_Type;
+      Length    : RFLX.TLV.Length_Type;
+      Value_Ref : RFLX.RFLX_Builtin_Types.Bytes_Ptr;
+
+   begin
+      Initialize (Ctx, 
+                  Buffer,
+                  Written_Last => 
+                     RFLX.RFLX_Types.To_Last_Bit_Index (Buffer'Last));
+ 
+      Verify_Message (Ctx);
+
+      if Well_Formed_Message (Ctx) then
+         Tag := Get_Tag (Ctx);
+         Put_Line ("Tag field is " & Tag'Img);
+         case Tag is
+            when RFLX.TLV.Msg_Data =>
+               Length := Get_Length (Ctx);
+               Put_Line( "Length field is " & Length'Img);
+               if Present (Ctx, F_Value) then
+                  Value_Ref := 
+                     new RFLX.RFLX_Types.Bytes'( Get_Value(Ctx) );
+                  Put ("Value field is ( ");
+                  for Byte_Val of Value_Ref.all loop
+                     Put( Byte_Val'Img & ' ');
+                  end loop;
+                  Put_Line ( ")" ); 
+               else
+                  Ada.Text_IO.Put_Line ("Data message without value");
+               end if;
+            when RFLX.TLV.Msg_Error =>
+               Ada.Text_IO.Put_Line ("Error message");
+         end case;
+      else
+         Ada.Text_IO.Put_Line ("Invalid message");
+      end if;
+   end Parse_TLV;
+
+* Lines 7-8
+
+  The message to be parsed is created as "raw storage" (a ``Bytes``
+  array). The ``Buffer`` variable points to an allocated copy of
+  the array.
+
+* Lines 15-18
+
+  ``Ctx`` is initialized with ``Buffer``, and ``Buffer'Last`` is
+  identified as the last element.
+
+* Line 20
+
+  .. TODO:: Explain  what ``Verify_Message`` does
+
+* Line 29
+
+  .. TODO:: Explain  why ``Present`` is called
```

Comment:
How about putting that explanations as Ada comments into the code block? I think that would make it easier for the reader, because there would be no need to search the corresponding code lines.